v_4_8_0/files/config/kubelet-configmap.yaml.tmpl

```diff
@@ -0,0 +1,80 @@
+apiVersion: v1
+kind: ConfigMap
+metadata: 
+  name: kubelet-configuration-master-values
+  namespace: kube-system
+data:
+  kubelet: |
```

Comment:
Do we have now to keep this and fallback configuration in sync? How is it done?

Reply:
No sync atm, they are different resources. If we are not fine with this we can reduce configmap configs to the minimum values required.

Reply:
It would be good to have at least bold comment in both places that those things need to be in sync.

Reply:
Yes, a comment would be the minimum but this is an interesting topic if it would be better to have a full kubelet config or just the overrides.

Reply:
I can't tell which is better.

Reply:
Maybe we should discuss in the SIG?

Reply:
My bad, you cannot select specific fields to override. A valid cm overrides a full config file.
https://github.com/giantswarm/opsctl/pull/409

Reply:
This brings some overhead. Why can't we at least render both configs from the same template?

Reply:
Do you mean refactor worker/master with same template passing different values depending if it is worker or master or creating a new template for configMap part and then there partially render the kubelet config template inside?

Reply:
I mean why can’t ypu create configmap from local kubelet config instead of rendering separate cm yaml?

Reply:
Understood I see the improvement there.
I will see how to integrate a template in the template and if it is possible.
 
```
apiVersion: v1
kind: ConfigMap
metadata: 
  name: kubelet-configuration-master-values
  namespace: kube-system
data:
  kubelet: |
```Render a template ehere```
```

Reply:
No, not render into cm
But create cm from file as we do for kube-proxy

kubectl create configmap kubelet --from-file=<local config>

You can add it into the kubelet config rendering service as a second command

Reply:
I got you now. Nice idea.
Thanks